packages/react-vapor/src/components/sideNavigation/SideNavigationItem.tsx

```diff
@@ -1,13 +1,42 @@
+import * as classNames from 'classnames';
 import * as React from 'react';
+import {addClassNameToChildren} from '../../utils/JSXUtils';
 
 export interface ISideNavigationItemProps {
     href: string;
     title: string;
     target?: string;
 }
 
-export const SideNavigationItem = (props: ISideNavigationItemProps) => (
-    <a className="block navigation-menu-section-item" href={props.href} target={props.target}>
-        <span className="navigation-menu-section-item-link">{props.title}</span>
-    </a>
-);
+export interface SideNavigationItemProps extends Partial<ISideNavigationItemProps> {
+    isActive?: boolean;
+}
+
+export const SideNavigationItem: React.FunctionComponent<SideNavigationItemProps> = ({
+    isActive,
+    href,
+    title,
+    children,
+    target,
+}) => {
+    const ref = React.useRef(null);
+
+    React.useEffect(() => {
+        if (isActive && ref.current) {
+            ref.current.scrollIntoView({behavior: 'instant', block: 'nearest'});
+        }
+    }, [isActive]);
+
+    const itemClasses = classNames('navigation-menu-section-item', {'state-active': isActive});
+
+    // Rendering an anchor tag from href and title support for retrocompatibility
+    return href && title ? (
+        <a className={classNames('block', itemClasses)} href={href} target={target} ref={ref}>
+            <span className="navigation-menu-section-item-link">{title}</span>

```

Comment:
I guess the title is right here right ?

Reply:
Here its the title of the SideNavigationItem, the other file is the title of the navigation section